job/server/src/main/java/alluxio/master/AlluxioJobMasterProcess.java

```diff
@@ -37,136 +37,97 @@
 import java.io.IOException;
 import java.net.InetSocketAddress;
 import java.net.ServerSocket;
-import java.util.Map;
-import java.util.concurrent.TimeoutException;
+import java.net.SocketAddress;
+import java.net.URI;
 
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.NotThreadSafe;
+import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * This class is responsible for initializing the different masters that are configured to run.
  */
 @NotThreadSafe
-public class AlluxioJobMasterProcess implements JobMasterProcess {
+public class AlluxioJobMasterProcess extends MasterProcess {
```

Comment:
should this implement `JobMasterProcess`?

Reply:
`JobMasterProcess` has been removed because I felt it added unnecessary complexity. Similar with the normal master as well. We don't have multiple implementations of each master. The `FaultTolerant` variants of each only extend the class and they don't override any methods which were required to be implemented by their respective original interfaces. They only were overriding methods from the `Process` interface.